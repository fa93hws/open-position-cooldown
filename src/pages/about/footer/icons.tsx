import * as React from 'react';

import weiboSVG from './weibo.svg';

function createIcon(svgContent: string) {
  return () => <i dangerouslySetInnerHTML={{ __html: svgContent }} />;
}

export const Weibo = createIcon(weiboSVG);
export type IconType = ReturnType<typeof createIcon>;
