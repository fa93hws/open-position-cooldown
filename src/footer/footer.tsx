import * as React from 'react';
import { Flex, Link } from 'rebass';
import { Github, Weibo, Code, IconType } from '../resources/icons';

const IconLink = (props: { Icon: IconType; href: string }) => (
  <Link
    size={['iconSmall', 'iconMedium']}
    p="0px"
    href={props.href}
    target="__blank"
    sx={{ fill: 'white' }}
  >
    <props.Icon />
  </Link>
);

export const Footer = () => (
  <Flex
    height={['50px', '70px']}
    bg="primary"
    justifyContent="space-around"
    alignItems="center"
    fontSize="0px"
    as="footer"
  >
    <IconLink Icon={Weibo} href="https://weibo.com/hinanawi" />
    <IconLink Icon={Github} href="https://github.com/fa93hws/" />
    <IconLink
      Icon={Code}
      href="https://github.com/fa93hws/open-position-cooldown"
    />
  </Flex>
);
