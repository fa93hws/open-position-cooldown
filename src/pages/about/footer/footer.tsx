import * as React from 'react';
import { Box, Link } from '@material-ui/core';
import { makeStyles, withTheme, WithTheme } from '@material-ui/core/styles';
import { GitHub, Code } from '@material-ui/icons';

import { Weibo } from './icons';
import { sizes } from '../../../styles/styles';

const iconSize = {
  width: sizes[1],
  height: sizes[1],
};
const useFooterStyle = makeStyles((theme) => ({
  container: {
    height: sizes[2],
    backgroundColor: theme.palette.primary.main,
  },
  iconWrapper: {
    color: theme.palette.primary.contrastText,
    fill: theme.palette.primary.contrastText,
    ...iconSize,
  },
  icon: {
    height: '100%',
    width: '100%',
  },
}));

export const Footer = withTheme((props: WithTheme) => {
  const styles = useFooterStyle(props.theme);
  return (
    <Box className={styles.container} display="flex" alignItems="center">
      <Box display="flex" justifyContent="space-around" flex="1">
        <Link
          href="https://github.com/fa93hws/"
          target="__blank"
          className={styles.iconWrapper}
        >
          {/* github icon is a bit larger than the others, add padding to reduce the size */}
          <GitHub className={styles.icon} style={{ padding: '4px' }} />
        </Link>
        <Link
          href="https://www.weibo.com/hinanawi/"
          target="__blank"
          className={styles.iconWrapper}
        >
          <Weibo />
        </Link>
        <Link
          href="https://github.com/fa93hws/open-position-cooldown"
          target="__blank"
          className={styles.iconWrapper}
        >
          <Code className={styles.icon} />
        </Link>
      </Box>
    </Box>
  );
});
