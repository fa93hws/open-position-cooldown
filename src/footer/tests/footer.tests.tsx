import * as React from 'react';

import { Footer } from '../footer';

describe('Footer', () => {
  it('renders it', () => {
    expect(<Footer />).toMatchRenderedSnapshot();
  });
});
