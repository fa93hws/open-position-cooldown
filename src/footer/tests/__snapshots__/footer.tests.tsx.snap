// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders it 1`] = `
<div
  class="css-tyqq70"
>
  <a
    class="css-1b61pal"
    href="https://weibo.com/hinanawi"
    style="fill:white"
    target="__blank"
  >
    <i />
  </a>
  <a
    class="css-1b61pal"
    href="https://github.com/fa93hws/"
    style="fill:white"
    target="__blank"
  >
    <i />
  </a>
  <a
    class="css-1b61pal"
    href="https://github.com/fa93hws/open-position-cooldown"
    style="fill:white"
    target="__blank"
  >
    <i />
  </a>
</div>
`;
